### PR TITLE
Social a-anxiety/st-st-stuttering t-trait.

### PR DESCRIPTION
**Pull request by VasilisThePikachu**
## About the PR

This pr adds a trait that makes you stutter when you speak, it uses stuttering component with the now ability to play around with its rng. Want more stutter? up the values!

## Why / Balance

It c-can provide some g-good r-roleplay int-taractions.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image

## Breaking changes

**Changelog**

:cl: Vasilis
- add: Added a s-social anxiety trait that m-makes you st-stutter your w-words when you t-talk.

### CODE REVIEW

**Comment by VasilisThePikachu** [comment]
Also i did check, it does not get removed after you are done being electrocuted/bleeding out/whatever else makes you stutter

**Comment by NancokPS2** [comment]
Could it maybe get toned down? This is making LOUD MOUTH seem ideal.

**Comment by LankLTE** [comment]
It should probably be toned down. The current stutter massively floods the chat box, but it’s not a huge issue because it isn’t 24/7 but I think having this available as a trait would cause headache for players and admins.

**Comment by deltanedas** [comment]
admin logs also say the origin message before accents

**Review by EmoGarbage404 — DISMISSED** [review]
tone down the accent. this makes chat borderline unreadable.

**Comment by Just-a-Unity-Dev** [comment]
I'd rather the actual ss13 trait where at times you randomly get mute

**Comment by VasilisThePikachu** [comment]
> I'd rather the actual ss13 trait where at times you randomly get mute

that how that works? thought it just made you stutter the more people and eventually muting you

**Comment by Just-a-Unity-Dev** [comment]
> > I'd rather the actual ss13 trait where at times you randomly get mute
> 
> that how that works? thought it just made you stutter the more people and eventually muting you

yes

**Comment by VasilisThePikachu** [comment]
> > > I'd rather the actual ss13 trait where at times you randomly get mute
> > 
> > 
> > that how that works? thought it just made you stutter the more people and eventually muting you
> 
> yes, also wdym by "muting you"

The one time I played 13 there were enough people around me to "lock me up" and was unable to speak until I left the area

**Comment by deltanedas** [comment]
my take is that speaking when nobody is around, including on radio,  it should be fine
then gradually stutter the more people there are that you can see, up to the unreadable stutter currently in a large croud

**Comment by JoeHammad1844** [comment]
absolutely unreadable and bloats the text box to nikocado avocado levels. 
imagine yelling for help on the radio only to get drowned out by someone who's sentance of text gets turned into a paragraph

**Comment by NancokPS2** [comment]
That's where the muting would come in handy.   
I wonder if that was the original intention.

**Comment by Warentan** [comment]
feels a bit overdone. Making it repeat only once would be better and less a hindrance to read...

**Comment by VasilisThePikachu** [comment]
Ok jesus christ why does this have 7 dislikes for an issue to be easily fixed my god people relax

**Comment by VasilisThePikachu** [comment]
Anyway added the ability to play with the rng chances, if you feel like it needs more stutter or less feel free to play with the values

**Comment by forgotmyotheraccount** [comment]
I'll be blunt that I don't like this trait and would rather not run into it at all; it's both aggravating to read and peak cringe (add "uwu" to the end of messages and it's the type of PR you'd expect to see pushed on an ERP server).

But if it really must be pushed, I'd ask that it not apply when the anxious person is whispering (aka whenever they use the radio)

**Comment by mirrorcult** [comment]
yeah i just really do not like this accent even toned down

**Comment by NancokPS2** [comment]
I don't see it much worse than lizard accent tbh.   
But it should only stutter on 1 of every 5 words tops.  
This is enough to have it be apparent but not actually make it harder to read.

File: Content.Server/Speech/Components/StutteringAccentComponent.cs [hunk 1]
```diff
@@ -3,5 +3,20 @@ namespace Content.Server.Speech.Components
...
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float FourRandomProb = 0.1f;
+
+        [DataField("threeRandomProb")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float ThreeRandomProb = 0.2f;
+
+        [DataField("cutRandomProb")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float CutRandomProb = 0.05f;
```
Comment (LankLTE) [inline Content.Server/Speech/Components/StutteringAccentComponent.cs:20]: These datafields need documentation on what they actually do.

**Review by LankLTE — CHANGES_REQUESTED** [review]

File: Content.Server/Speech/Components/StutteringAccentComponent.cs (hunk 1 above)
Comment (VasilisThePikachu) [reply Content.Server/Speech/Components/StutteringAccentComponent.cs]: Ya got it

**Comment by JoeHammad1844** [comment]
this would just bloat text and annoy people

**Comment by Tunguso4ka** [comment]
would love to see it.

**Comment by Eden077** [comment]
Looks much better toned down

**Review by LankLTE — APPROVED** [review]
Looks good, sure.

**Comment by DummeHytteOst** [comment]
i have gotten a feel for this trait after having used it for some hours now, and i feel like you stutter way too much, and way too often, it gets unreadable at times and becomes a wall of text, i had a simple "ah" become "ah-h-h-h", sometimes it becomes almost double the text with the stutter in some cases.  

i suggest lowering the chance to stutter.

**Comment by VasilisThePikachu** [comment]
yeah, that's one thing I suck at, balancing numbers. Feel free to pr it yourself. It's easy to modify

**Pull request merged**